tests/rptest/tests/cluster_config_test.py

```diff
@@ -1561,9 +1564,32 @@ def _get_prop_set(self, name: str) -> PropertyAliasData:
         self.logger.info(f"prop_set={vars[name]}")
         return vars[name]
 
+    @cluster(num_nodes=1)
+    @matrix(prop_set_str=['log_retention_ms'])
+    def test_deprecation_of_alias(self, prop_set_str: str):
+        """
```

Comment:
Can we also add a test where in an older version, we set `delete_retention_ms`, and then upgrade? I guess the expected behavior is that the config doesn't carry over? Which seems a bit scary... Maybe worth double checking the expected behavior in that case

Reply:
The config carries over. test_aliasing_with_upgrade, a bit after this, shows the upgrade behavior. Or do you have in mind another sequence of actions/checks?

Reply:
Specially I'm wondering about the case where:
- in a version that only has `delete_retention_ms`, we set it to some non-default value 30d
- then we upgrade to 24.1, and the admin doesn't touch anything. Redpanda correctly thinks `log_retention_ms` is 30d
- then we upgrade to 24.2, and the admin doesn't touch anything again. At this point, Redpanda no longer thinks `delete_retention_ms` is no longer an alias of `log_retention_ms`. Does the config carry over?

I see similar steps in `test_aliasing_with_upgrade`, but in that test we also reset the config in between upgrades. I'm wondering whether this behaves as expected if we _don't_ do those config resets.

Reply:
right, i see now. 

the translation mechanism is in [config_manager::store_delta](https://github.com/redpanda-data/redpanda/blob/4458d3be3bb2f0e77428ad3f9bcc3699afe03537/src/v/cluster/config_manager.cc#L814-L845) and it's independent from the set endpoint, 
I added one last commit to show that.

But you are right that it's not a perfect system: store_delta gets a cmd from the controller log, then saves the primary name in the raw_values map, which will fill the `config_cache.yaml` immediately and the snapshot when truncation happens. 

This means that if 
1. config_cache.yaml is deleted, then 
2. the cluster upgraded before snapshot + truncation can happen
then the new version has no way to know how to handle the now-deprecated delete_retention_ms and log_retention_ms will have the default cluster value.
This will appear in the failure, hopefully in a couple of hours or so.